use_case_examples/resnet/run_resnet18_fhe.py

```diff
@@ -53,7 +54,12 @@ def evaluate_model(model, processor):
 
 
 def compile_model(
-    model, images, n_bits, rounding_threshold_bits=None, fhe_mode="disable", use_gpu=False
+    model,
+    images,
+    n_bits,
+    rounding_threshold_bits=None,
+    fhe_mode="disable",
+    compilation_device="cpu",
```

Comment:
I preferred `use_gpu` or just `device=...` to do the same as torch.

Reply:
"@jfrery, `use_device` might sound like a boolean type, but in the compilation functions, it's actually a string that can be either 'cpu' or 'gpu'. 

The reason it's not simply named `device` is because the PyTorch device and the CML one may differ. This distinction helps differentiate between the device used for PyTorch training/evaluation and the one used for CML compilation.

Reply:
@jfrery the API was decided some time ago:

- https://zama-ai.slack.com/archives/C0455QDGT6C/p1723040023589979
- https://zama-ai.slack.com/archives/C0455QDGT6C/p1724860575912049

Reply:
`device="cuda|cpu"` is perfect yes but here we have compilation_device. 

> The reason it's not simply named device is because the PyTorch device and the CML one may differ.

Well as far as I know we try to stay close to the torch API. We did it for sklearn and the others. I don't see why now we would want to differ. device seems the right choice here.

Reply:
We assume that the compile_device may differ from the device used to train the model.

Additionally, on standard machines, setting compile_device="cpu" is currently faster than compile_device="cuda". So, it's better to keep them separate.